hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryEndpointWithLegacy.java

```diff
@@ -0,0 +1,1286 @@
+    Assert.assertEquals(VOL_DATA_SIZE, quVolRes.getQuotaUsed());
+
+    // bucket level quota usage
+    Response bucketRes = nsSummaryEndpoint.getQuotaUsage(BUCKET_ONE_PATH);
+    QuotaUsageResponse quBucketRes = (QuotaUsageResponse) bucketRes.getEntity();
+    Assert.assertEquals(BUCKET_ONE_QUOTA, quBucketRes.getQuota());
+    Assert.assertEquals(BUCKET_ONE_DATA_SIZE, quBucketRes.getQuotaUsed());
+
+    Response bucketRes2 = nsSummaryEndpoint.getQuotaUsage(BUCKET_TWO_PATH);
+    QuotaUsageResponse quBucketRes2 =
+        (QuotaUsageResponse) bucketRes2.getEntity();
+    Assert.assertEquals(BUCKET_TWO_QUOTA, quBucketRes2.getQuota());
+    Assert.assertEquals(BUCKET_TWO_DATA_SIZE, quBucketRes2.getQuotaUsed());
+
+    // other level not applicable
+    Response naResponse1 = nsSummaryEndpoint.getQuotaUsage(DIR_ONE_PATH);
+    QuotaUsageResponse quotaUsageResponse1 =
+        (QuotaUsageResponse) naResponse1.getEntity();
+    Assert.assertEquals(ResponseStatus.TYPE_NOT_APPLICABLE,
+        quotaUsageResponse1.getResponseCode());
+
+    Response naResponse2 = nsSummaryEndpoint.getQuotaUsage(KEY_PATH);
+    QuotaUsageResponse quotaUsageResponse2 =
+        (QuotaUsageResponse) naResponse2.getEntity();
+    Assert.assertEquals(ResponseStatus.TYPE_NOT_APPLICABLE,
+        quotaUsageResponse2.getResponseCode());
+
+    // invalid path request
+    Response invalidRes = nsSummaryEndpoint.getQuotaUsage(INVALID_PATH);
+    QuotaUsageResponse invalidResObj =
+        (QuotaUsageResponse) invalidRes.getEntity();
+    Assert.assertEquals(ResponseStatus.PATH_NOT_FOUND,
+        invalidResObj.getResponseCode());
+  }
+
+
+  @Test
+  public void testFileSizeDist() throws Exception {
+    checkFileSizeDist(ROOT_PATH, 2, 3, 4, 1);
+    checkFileSizeDist(VOL_PATH, 2, 1, 2, 1);
+    checkFileSizeDist(BUCKET_ONE_PATH, 1, 1, 1, 1);
+    checkFileSizeDist(DIR_ONE_PATH, 0, 1, 1, 1);
+  }
+
+  public void checkFileSizeDist(String path, int bin0,
+      int bin1, int bin2, int bin3) throws Exception {
+    Response res = nsSummaryEndpoint.getFileSizeDistribution(path);
+    FileSizeDistributionResponse fileSizeDistResObj =
+            (FileSizeDistributionResponse) res.getEntity();
+    int[] fileSizeDist = fileSizeDistResObj.getFileSizeDist();
+    Assert.assertEquals(bin0, fileSizeDist[0]);
+    Assert.assertEquals(bin1, fileSizeDist[1]);
+    Assert.assertEquals(bin2, fileSizeDist[2]);
+    Assert.assertEquals(bin3, fileSizeDist[3]);
+    for (int i = 4; i < ReconConstants.NUM_OF_BINS; ++i) {
+      Assert.assertEquals(0, fileSizeDist[i]);
+    }
+  }
+
+  /**
+   * Write directories and keys info into OM DB.
+   * @throws Exception
+   */
+  @SuppressWarnings("checkstyle:MethodLength")
+  private void populateOMDB() throws Exception {
+    // write all directories
+    writeDirToOm(reconOMMetadataManager,
+          (DIR_ONE + OM_KEY_PREFIX),
+          BUCKET_ONE,
+          VOL,
+          DIR_ONE,
+          DIR_ONE_OBJECT_ID,
+          BUCKET_ONE_OBJECT_ID,
```

Comment:
The parent object id for legacy records is 0, (which is why we added the setKeyParentId() method.) Let's set them back to 0 in the test code as well, just to be sure that setKeyParentId() is really working.